nix/nim-status/default.nix

```diff
@@ -0,0 +1,150 @@
+{ newScope
+, mkShell
+, writeScript
+, writeText
+, xcodeWrapper
+, pkgs
+, stdenv
+, lib }:
+let
+  callPackage = newScope {};
+  buildNimStatus = platform: arch: callPackage ./nim-status.nix {
+    inherit platform arch;
+  };
+
+  # Metadata common to all builds of status-go
+  meta = {
+    description = "The Status Go module that consumes go-ethereum.";
+    license = lib.licenses.mpl20;
+    platforms = with lib.platforms; linux ++ darwin;
+  };
+
+  # Source can be changed with a local override from config
+  source = callPackage ./status-go-source.nix { };
+  
+  buildStatusGo = platform: arch: callPackage ./status-go.nix {
+    inherit platform arch meta source;
+  };
+
+  buildAndroid = buildMap: name: stdenv.mkDerivation {
+    name = "${name}-android-builder";
```

Comment:
Why are you calling this a builder? The script you pass via `writeScript` into `builder` IS a builder. But the resulting derivation form running the builder is NOT a builder, it's the package itself.

Reply:
Changed the names.